Source/Extensions/Blazorise.Charts/BaseChartPlugin.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Threading.Tasks;
+using Blazorise.Extensions;
+using Blazorise.Modules;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace Blazorise.Charts;
+
+public abstract class BaseChartPlugin : BaseComponent, IAsyncDisposable
+{
+    protected internal abstract Task OnParentChartInitialized();
+    protected internal abstract string Name { get; }
+}
+
+public abstract class BaseChartPlugin<TItem, TJSModule> : BaseChartPlugin
+where TJSModule : IBaseJSModule, IAsyncDisposable
+{
+
+    protected abstract TJSModule GetNewJsModule();
+
+    protected abstract Task AddPlugin();
+
+    protected abstract bool ProceedWithExecution( ParameterView parameterView );
```

Comment:
ProceedWithExecution name seems strange. Execution of what? Might need better name for future clarity.

Reply:
It is weird, yeah, I will do something about it.